src/ansys/mechanical/core/embedding/app.py

```diff
@@ -115,7 +115,13 @@ def new(self):
         self.DataModel.Project.New()
 
     def close(self):
-        """Close the application."""
+        """Close the application and remove the lock file."""
+        lockfile_path = os.path.join(self.DataModel.Project.ProjectDirectory, ".mech_lock")
+        if os.path.exists(lockfile_path):
+            try:
+                os.remove(lockfile_path)
+            except:
+                raise Exception("There was an issue removing the lock file.")
         self.ExtAPI.Application.Close()
```

Comment:
What about this close operation? If an error is raised, do you still want to call close? Seems to me that if now an error is encountered when deleting the lock file (if it exists)... we won't be able to ever close the app.

I would consider adding this statement right before the "raise" as well, inside the "except"

Reply:
Yes, we should close the app either way. Thanks!

Reply:
@klmcadams I think we should remove the lock file _after_ closing, but ideally Close() itself should remove the lock file.  Can you start an email thread with Lei Gong about this?